src/util/StreamerMode.cpp

```diff
@@ -32,51 +42,109 @@ bool isInStreamerMode()
             return true;
         case StreamerModeSetting::Disabled:
             return false;
-    }
+        case StreamerModeSetting::DetectObs:
 
-#ifdef USEWINSDK
-    if (!IsWindowsVistaOrGreater())
-    {
-        return false;
-    }
-    static bool cache = false;
-    static QDateTime time = QDateTime();
+#ifdef Q_OS_LINUX
 
-    if (time.isValid() &&
-        time.addSecs(cooldownInS) > QDateTime::currentDateTime())
-    {
-        return cache;
-    }
+            static bool cache = false;
+            static QDateTime time = QDateTime();
+
+            if (time.isValid() &&
+                time.addSecs(cooldownInS) > QDateTime::currentDateTime())
+            {
+                return cache;
+            }
+            time = QDateTime::currentDateTime();
 
-    time = QDateTime::currentDateTime();
+            QProcess p;
+            p.start("pgrep", {"-x", broadcastingBinaries().join("|")},
+                    QIODevice::NotOpen);
 
-    WTS_PROCESS_INFO *pWPIs = nullptr;
-    DWORD dwProcCount = 0;
+            if (p.waitForFinished(1000) &&
+                p.exitStatus() == QProcess::NormalExit)
+            {
+                cache = (p.exitCode() == 0);
+                return (p.exitCode() == 0);
+            }
 
-    if (WTSEnumerateProcesses(WTS_CURRENT_SERVER_HANDLE, NULL, 1, &pWPIs,
-                              &dwProcCount))
-    {
-        //Go through all processes retrieved
-        for (DWORD i = 0; i < dwProcCount; i++)
-        {
-            QString processName = QString::fromUtf16(
-                reinterpret_cast<char16_t *>(pWPIs[i].pProcessName));
+            // Fallback to true and showing a warning
 
-            if (broadcastingBinaries().contains(processName))
+            if (shouldShowWarning)
             {
-                cache = true;
-                return true;
+                shouldShowWarning = false;
+
+                if (auto selected = getApp()
+                                        ->windows->getMainWindow()
+                                        .getNotebook()
+                                        .getSelectedPage())
+                {
+                    if (auto container =
+                            dynamic_cast<SplitContainer *>(selected))
+                    {
+                        for (auto &&split : container->getSplits())
+                        {
+                            if (auto channel = split->getChannel();
+                                !channel->isEmpty())
+                            {
+                                channel->addMessage(makeSystemMessage(
+                                    "Streamer mode got enabled, due to pgrep "
+                                    "missing. Install it to fix the issue."));
```

Comment:
When the "default on error" value has been reverted back to false, this phrasing needs to update too.
Maybe something along the lines of:
"Streamer mode is set to automatic, but pgrep is missing. Install it to fix the issue or set streamer mode to enabled or disabled manually in the settings."